<a href="https://colab.research.google.com/github/Vkm3221/Assignments-Hypothesis-Testing/blob/main/Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import scipy.stats as stats
import statsmodels.api as sm
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings("ignore")
from PIL import ImageGrab
import matplotlib.pyplot as plt
import seaborn as sns

Q-01: A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomlyselected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Pleasestate the assumptions and tests that you carried out to check validity of the assumptions




Null Hypothesis:  u1=u2

Alternative Hypthosis: u1 not equal to u2


Decide a cut-off value

Significance 5%
alpha = 0.05

As it is a two-tailed test
alpha/2 = 0.025






In [ ]:
cutlets = pd.read_csv('/content/Cutlets.csv')
cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
cutlets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [ ]:
cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [ ]:
cutlets.isnull().sum()

Unit A    0
Unit B    0
dtype: int64

In [ ]:
stats.ttest_ind(cutlets['Unit A'],cutlets['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [ ]:
p_value= stats.ttest_ind(cutlets['Unit A'],cutlets['Unit B'])[1]
p_value

0.4722394724599501

Here, significance= 0.025 and p-value= 0.472
p-value greater than Significnace. So, we fail to reject the null hypothesis.

Q-02: A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level


Here, 4 indpendent sample numerical data, so apply anova test

Null Hypothesis: u1=u2=u3=u4

Alternative Hypthosis: Atleast One of them is Differente

Significance 5%

alpha = 0.05

In [ ]:
LabTAT=pd.read_csv('/content/LabTAT.csv')
LabTAT.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [ ]:
LabTAT.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory 1  120 non-null    float64
 1   Laboratory 2  120 non-null    float64
 2   Laboratory 3  120 non-null    float64
 3   Laboratory 4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB


In [ ]:
LabTAT.isnull().sum()

Laboratory 1    0
Laboratory 2    0
Laboratory 3    0
Laboratory 4    0
dtype: int64

In [ ]:
LabTAT.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [ ]:
stats.f_oneway(LabTAT.iloc[:,0],LabTAT.iloc[:,1],LabTAT.iloc[:,2], LabTAT.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [ ]:
p_value= stats.f_oneway(LabTAT.iloc[:,0],LabTAT.iloc[:,1],LabTAT.iloc[:,2], LabTAT.iloc[:,3])[1]
p_value

2.1156708949992414e-57

Since the p-value here is lesser than 0.05, we shall conclude that the Alternate Hypothesis is true, which says that atleast One of them is Differente

Q. no.-03 Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

**Using Chi-Square test with Contingency table**


*   Null Hypothesis: There is no association or dependency between the gender based buyer rations across regions

*   Alternative Hypthosis: There is a significant association or dependency between the gender based buyer rations across regions

Significance 5%

alpha = 0.05



In [ ]:
df= pd.read_csv('/content/BuyerRatio.csv')
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
df_table=df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
df_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
val=stats.chi2_contingency(df_table)
val

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [ ]:
Expected_value=val[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [ ]:
observed = np.array([50, 142, 131, 70, 435, 1523, 1356, 750])
expected = np.array([42.76531299,  146.81287862,  131.11756787, 72.30424052, 442.23468701, 1518.18712138, 1355.88243213, 747.69575948])

In [ ]:
statistics, p_value = stats.chisquare(observed, expected, ddof = 3)
print("Statistics = ",statistics,"\n",'P_Value = ', p_value)

Statistics =  1.5959455390914483 
 P_Value =  0.8095206646905712


Since the p-value here is greater than 0.05, we shall conclude that We fail to reject Null Hypothesis because of lack evidence. Therefore, there is no association or dependency between male-female buyers rations and are similar across regions. Hence, Independent samples

*****************************************************************************************

Q. No-04 TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level 

Using Chi-Square test with Contingency table

Null Hypothesis:u1=u2=u3=u4

Alternative Hypthosis: Atleast One of them is Differente

Significance 5%

alpha = 0.05




In [ ]:
df= pd.read_csv('/content/Costomer+OrderForm.csv')
df

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [ ]:
df.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [ ]:
df.isnull().sum()

Phillippines    0
Indonesia       0
Malta           0
India           0
dtype: int64

In [ ]:
print(df['Phillippines'].value_counts(),'\n',df['Indonesia'].value_counts(),
      '\n',df['Malta'].value_counts(),'\n',df['India'].value_counts)

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 
 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 
 Error Free    269
Defective      31
Name: Malta, dtype: int64 
 <bound method IndexOpsMixin.value_counts of 0      Error Free
1       Defective
2      Error Free
3      Error Free
4      Error Free
          ...    
295    Error Free
296    Error Free
297    Error Free
298    Error Free
299    Error Free
Name: India, Length: 300, dtype: object>


In [ ]:
observed = np.array([271, 267, 269, 280, 29, 33, 31, 20])
expected = np.array([271.75, 271.75, 271.75, 271.75, 28.25, 28.25, 28.25, 28.25])

In [ ]:
statistics, p_value = stats.chisquare(observed, expected, ddof = 3)
print("Statistics = ",statistics,"\n",'P_Value = ', p_value)

Statistics =  3.858960685820355 
 P_Value =  0.4254298144535761


Since the p-value here is greater than 0.05, we shall conclude that We fail to reject Null Hypothesis